src/linked_hash_map.rs

```diff
@@ -1344,6 +1389,33 @@ pub struct Drain<'a, K, V> {
     marker: PhantomData<(K, V, &'a LinkedHashMap<K, V>)>,
 }
 
+/// The `CursorMut` struct and its implementation provide the basic mutable Cursor API for Linked
+/// lists as proposed in
+/// [here](https://github.com/rust-lang/rfcs/blob/master/text/2570-linked-list-cursors.md), with
+/// several exceptions:
+///
+/// - It behaves similarly to Rust's Iterators, returning `None` when the end of the list is
+///   reached. A _guard_ node is positioned between the head and tail of the linked list to
+///   facilitate this. If the cursor is over this guard node, `None` is returned, signaling the end
+///   or start of the list. From this position, the cursor can move in either direction as the
+///   linked list is circular, with the guard node connecting the two ends.
+/// - The current implementation does not include an `index` method, as it does not track the index
+///   of its elements. It operates by providing items as key-value tuples, allowing the value to be
+///   modified via a mutable reference while the key could not be changed.
+/// - The current implementation does not include `splice_*`, `split_*`, `as_cursor`, or
+///   `remove_current` methods, as there hasn't been a strong demand for these features in
+///   real-world scenarios. However, they can be readily incorporated into the existing codebase if
+///   needed.
+/// - For added convenience, it includes the `move_at` method.
+///
```

Comment:
I think we can maybe drop this part of the doc comments?

Reply:
Sounds good to me.